### DISCUSSION TITLE
PopupBox is not running RippleAssist

### DISCUSSION THREAD

**Discussion by Frillaut** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3196] [Q&A] [answered]
Good afternoon!
This code doesn't change anything. What needs to be done, what would it work?

```
<md:PopupBox md:RippleAssist.ClipToBounds="True" md:RippleAssist.IsCentered="False"/>
```

**Comment by nicolaihenriksen** [comment]
**UPDATE:** The `RippleAssist.ClipToBounds` and `RippleAssist.IsCentered` are actually defined with `FrameworkPropertyMetadataOptions.Inherits` so they don't need to be explicitly passed down (I have "removed" parts of my comment below). But the hardcoding of the values would still need to change for this to work.

@Frillaut You are correct that those properties currently have no effect. ~~The values applied on the `PopupBox` are not passed down to the nested `Ripple` control~~. They are simply hardcoded as you can see [here](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/d2530d6fb4f4930d51c365ec53b7425993060f5e/MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.PopupBox.xaml#L108-L109) and [here](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/d2530d6fb4f4930d51c365ec53b7425993060f5e/MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.PopupBox.xaml#L271-L273).

Whether that is a deliberate decision I really don't know. It's a fairly trivial task to enable this functionality though. Feel free to open a PR with the changes if you have the time. I think @Keboo can then decide whether he thinks it "should be allowed" to control those properties 😃

**Reply by Frillaut** [reply]
Hello!
I'm new here (on github) and don't know how to open PR yet.
On the question. Yes, I saw that it was hard-coded. I hoped there could be a way in which I would not transform the whole style.

**Reply by nicolaihenriksen** [reply, marked as answer]
No problem. I have added a [PR](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/pull/3198) with the suggestion.

Without these changes, I think your only option is to do a full copy-paste of the template.

**Reply by nicolaihenriksen** [reply]
@Frillaut The PR merged, so you could try out the latest nightly build.

**Reply by Frillaut** [reply]
I looked. This property **md:RippleAssist.IsCentered="False"** still does not change anything. This is not critical for me, but still, I think, you wanted to make working and this property.

**Reply by Frillaut** [reply]
Works if you add a parent binding to Ripple
`md:RippleAssist.IsCentered="{TemplateBinding md:RippleAssist.IsCentered}"`

**Reply by nicolaihenriksen** [reply]
@Frillaut I can see that there hasn't been a nightly release pushed out yet. So that is probably why it doesn't work.
